tests/taker_cli.rs

```diff
@@ -0,0 +1,241 @@
+use bitcoin::{address::NetworkChecked, Address, Amount};
+use bitcoind::{bitcoincore_rpc::RpcApi, tempfile::tempdir, BitcoinD, Conf};
+
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+    str::FromStr,
+};
+
+/// The taker-cli command struct
+/// Use it to perform all taker-cli operations
+struct TakerCli {
+    /// taker-cli binary path
+    target: String,
+    rpc_network: String,
+    /// Connection type
+    network: String,
+    data_dir: Option<PathBuf>,
+    /// Bitcoind instance
+    bitcoind: BitcoinD,
+    fee_rate: u64,
+    /// no of makers in coinswap
+    makers_count: usize,
+    /// no of funding tx created in each swap
+    tx_count: u32,
+}
+
+impl TakerCli {
+    /// Construct a new [`TakerCli`] struct that also include initiating bitcoind.
+    fn new(
+        data_dir: Option<PathBuf>,
+        rpc_network: &str,
+        network: &str,
+        fee_rate: u64,
+        makers_count: usize,
+        tx_count: u32,
+    ) -> TakerCli {
+        // Initiate bitcoind
+        let temp_dir = tempdir().unwrap().into_path();
+
+        let mut conf = Conf::default();
+
+        conf.args.push("-txindex=1"); //txindex is must, or else wallet sync won't work.
+        conf.staticdir = Some(temp_dir.join(".bitcoin"));
+
+        let key = "BITCOIND_EXE";
+        let curr_dir_path = std::env::current_dir().unwrap();
+        let bitcoind_path = curr_dir_path.join("bin").join("bitcoind");
```

Comment:
For macos you will need the other `bitcoind_macos` binary. So this should have a feature gate, if the build env is mac then use macos binary, or use the `bitcoind` path.